### ISSUE TITLE
Media Field random image order

### ISSUE THREAD

**Issue by cwojke** [OrchardCMS/OrchardCore#1749]
If you have multiple images in Meida Field, images does not retain your sort order, you have to reorder them before each publish/save. The order is saved fine, but when the Content Item Edit page reloads, they are displayed randomly. The script **mediafield.js** is checking them whether they exits via ajax and that is causing this error. 

I think this part of code causes the problem (OrchardCore.Media\Assets\js\mediafield.js, line 23-43)

```
                set: function (values) {
                    var self = this;
                    var mediaPaths = values || [];
                    var signal = $.Deferred();
                    mediaPaths.forEach(function (x, i) {
                        promise = $.when(signal).done(function () {
                            $.ajax({
                                url: mediaItemUrl + "?path=" + encodeURIComponent(x),
                                method: 'GET',
                                success: function (data) {
                                    self.mediaItems.push(data);
                                },
                                error: function (error) {
                                    console.log(JSON.stringify(error));
                                }
                            });
                        });
                    });

                    signal.resolve();
                }
```

**Comment by matiasmolleja** [comment]
By reading the code I think you are right. But I can't repro :) 
For me the order is always OK. What steps are you following to make it fail?

**Comment by cwojke** [comment]
Maybe it's because I load images from Azure Blob Storage. I'm going to try it tomorrow with default media provider and I will see if it's behave correctly.

**Comment by matiasmolleja** [comment]
Closing after testing in azure blobstorage. Fixed by #1751
